server/src/main/java/org/opensearch/common/util/concurrent/TimedRunnable.java

```diff
@@ -107,6 +107,14 @@ long getTotalExecutionNanos() {
         return Math.max(finishTimeNanos - startTimeNanos, 1);
     }
 
+    long getWaitTimeNanos() {
```

Comment:
I guess we should create a follow-up to see if all the threadpool runnables can be wrapped behind TimedRunnable. It will help to get these time related stats which can be useful to understand if a threadpool type is over used across use cases (something like generic threadpool ?).

Reply:
Yep will create a follow-up task for this

Reply:
Follow-up task: https://github.com/opensearch-project/OpenSearch/issues/9983